### ISSUE TITLE
Use a better background color on full-line changes for colorblind users

### ISSUE THREAD

**Issue by ahatzz11** [ForkIssues/Tracker#785]
![image](https://user-images.githubusercontent.com/6256032/67304881-1e3e3b00-f4ba-11e9-8aa5-1975e637c26d.png)

The color difference in this red/green is almost impossible to see for colorblind users, especially with the dark theme. 

Here is an example of another git-client that has some better colors: 

![image](https://user-images.githubusercontent.com/6256032/67305000-4e85d980-f4ba-11e9-9b41-05b08db70a71.png)

I saw an issue previously that said it was hard to implement custom colors, which is fine, but tweaking the defaults would be excellent.

**Comment by DanPristupov** [comment]
We use same colors as GitHub.

**Comment by DanPristupov** [comment]
I think now we can introduce a custom colors for the High Contrast mode in MacOS (starting from 10.14 Mojave)

<img width="672" alt="Screenshot 2019-10-22 at 18 11 24" src="https://user-images.githubusercontent.com/618115/67306516-62036580-f4f7-11e9-8903-810446cd6822.png">

**Comment by ahatzz11** [comment]
I don't use a dark github for this exact reason! I've always wished they had better colors. Unfortunately I don't have mojave (and probably won't for a while because my updates are controller by the enterprise).

**Comment by ahatzz11** [comment]
If custom colors are difficult, maybe a second color option within the app would be more reasonable?

**Comment by DanPristupov** [comment]
Now macOS already has a feature to manage colors and we are going to migrate to it completely soon.
I'm planning to remove the dark theme support from pre-Mojave versions because it's built on hacks and I don't have an ability to test it anymore.

**Comment by photomoose** [comment]
It would be nice if we could customise the diff colours within the app rather than having to rely on the macOS features. Although the colour blind options in macOS work, it also distorts the colours in all my other applications, which is something I don't want to happen.

**Comment by laurentvaills** [comment]
+1 to have more suitable colours on the diff with the dark theme, because the current ones are really hard for colorblind users. I wanted to report that issues but saw that it was already reported quite few times. 

I have 2.0.4 and still the difference of color is not good like in this screenshot:
![image](https://user-images.githubusercontent.com/724346/99991492-5e56a800-2db5-11eb-8bfd-f467ccae26d0.png)

A `+` and `-` on the lines would actually be very useful . 
Or an option "never use dark theme" (I can see there is "Always use dark theme").

**Comment by danhowelldev** [comment]
I'm another one of the 8% of males and .5% of females that are red-green colorblind.

Please please consider making the contrast between the colors greater. The GitHub dark view is also not accessible and should not be used as a standard. And also agree with the comment above about adding `+` and `-`  to the lines.

Something like this (I just maxed the red and green in GitHub diff view)

OLD:
![Screen Shot 2021-03-26 at 4 26 37 pm](https://user-images.githubusercontent.com/12741800/112586595-31440000-8e50-11eb-97f2-7db1045ae4b0.png)

Change (likewise for green):
![Screen Shot 2021-03-26 at 4 29 15 pm](https://user-images.githubusercontent.com/12741800/112586778-7a944f80-8e50-11eb-8113-9b8089e6ea25.png)

NEW:
![Screen Shot 2021-03-26 at 4 26 01 pm](https://user-images.githubusercontent.com/12741800/112586632-4456d000-8e50-11eb-8781-c9e2a099907c.png)

**Comment by photomoose** [comment]
91.5% of developers are male (https://www.statista.com/statistics/1126823/worldwide-developer-gender/) - and 8% of males are colourblind - so assuming averages, this works out to be about 7% of your user base.

You've got a great app, I really like it - but the $49.99 licence is inconceivable to me if I cannot see the diffs.

Please consider adding the "+/-" clues as suggested by @danhowelldev as a bare minimum, then work on the colour customisation - or at least choose shades that are easily distinguishable. I don't care what the colours are, as long as there is an obvious difference in colour/contrast. 

Try running the app through colourblind simulators to simulate what we see:

<img width="1237" alt="Screenshot 2021-03-26 at 18 40 02" src="https://user-images.githubusercontent.com/1983638/112678459-db2c8800-8e62-11eb-90cb-b205fa7d14ed.png">

**Comment by DanPristupov** [comment]
Fork will use a special color scheme if you enable High Contrast mode in the system preferences:

<img width="400" alt="Screenshot 2021-03-26 at 19 50 08" src="https://user-images.githubusercontent.com/618115/112679456-7bd37580-8e6c-11eb-8c85-20ce88d5233c.png">

**Comment by photomoose** [comment]
@DanPristupov Could you introduce an option to enable this special colour scheme without having to change the OS system preferences? I don't really want to use the High Contrast mode for everything; it's just Fork where I need to distinguish colours.

**Comment by DanPristupov** [comment]
No, I can't. This is a macOS feature.

**Comment by danhowelldev** [comment]
I'm really confused. You keep just discarding these concerns and saying we need to use the high contrast mode. That's ridiculous, I don't want to see all my applications in some glaring high contrast view. Please just change the default colours.

**Comment by bennyty** [comment]
This is my number 1 issue with Fork. Literally every other thing about your app I *love*. But I frequently struggle to tell if an isolated change is an addition or subtraction.

I understand you're probably using Apple's auto-switching color pallets. That's why it was easy to add a high contrast mode but hard to have an app specific toggle for the color scheme.

How about this option mentioned above. Other apps have dark modes that have correct & usable contrast so setting an OS level dark mode makes sense. However Fork's contrast is too low so I'd like to disable Fork's dark theme. Instead of the "always dark" checkbox: Light, Dark, or Auto (match OS).

> Or an option "never use dark theme" (I can see there is "Always use dark theme").

**Comment by EvCholesky** [comment]
+1 vote for another user who really enjoys using your app but cannot tell the difference between the add and remove colors.

**Comment by photomoose** [comment]
@EvCholesky if you enable the "Show -+ marks" setting in Preferences -> General -> Diff View, it's a great help. Colours still suck though.

**Comment by kevinrenskers** [comment]
Today I decided to try out Fork after seeing some positive messages on Mastodon. And yeah.. it's basically not usable for me either.

Please use SourceTree's color, as they are much much better:

![Screenshot 2023-07-04 at 22 06 28](https://github.com/fork-dev/Tracker/assets/229384/59347687-4c6d-46e6-9d0f-7a257cc2cd35)

Compared to Fork:

![Screenshot 2023-07-04 at 22 07 15](https://github.com/fork-dev/Tracker/assets/229384/76739d5b-50c9-4ede-aabf-9f9679099e77)

If you're not colorblind then you probably don't get what the fuss is about, but trust us: changing the colors would be a huge difference for us. Until then I can't use this app and won't be paying for it.

**Comment by joki-1122** [comment]
I'm experiencing the same issue—these colors make it unusable.

**Comment by ovenpasta** [comment]
This is crazy even though I can see the colors properly, with night light Windows setting it becomes unusable. Please allow to customize colors. Otherwise great app!

**Comment by NinoScript** [comment]
Hey, I purchased Fork, but I have the same problem as everybody else in this issue.
If you don't want to give us a way to change the colors, please just use a "bluer" kind of green. Use the colors you would see in a traffic light, or the ones they're showing in sourcetree. Those work much better for us.

**Comment by DanPristupov** [comment]
@NinoScript 

>  Use the colors you would see in a traffic light, or the ones they're showing in sourcetree. Those work much better for us.

If you enable high contrast mode, they will be used https://github.com/fork-dev/Tracker/issues/785#issuecomment-808443274

**Comment by NinoScript** [comment]
I understand, but I cannot use high contrast because of my light sensitivity.
This is not about contrast, this is about hue. High contrast mode is for another set of people.

**Comment by photomoose** [comment]
Agreed. We don’t want system wide high contrast (which affects every other
app), that’s not the issue. It’s the colour that’s the problem. Have you
tried using a colour blind simulator to see how it looks to colourblind
people?

Some apps (e.g. Slack) have a colourblind theme which changes the hues
used.

**Comment by NinoScript** [comment]
Also, you cannot solve this by just having a single colorblind setting, as there are multiple types of colorblindness.

You said you use the same colors as GitHub.
GitHub is aware of this problem with the colors they chose, and they provide two extra themes:
* "colorblind": for people with Deuteranopia or Protanopia (this is the one I use)
* "tritanopia": for people with Tritanopia (blue-yellow colorblindness)

<img width="960" height="840" alt="Image" src="https://github.com/user-attachments/assets/dcdfe574-aedd-4c7f-a2a5-0a222b25d395" />

In case it's useful, they have their theme's colors in the [@primer/primitives](https://github.com/primer/primitives) package.
In the [`@primer/primitives/dist/css/functional/themes/` directory](https://www.npmjs.com/package/@primer/primitives?activeTab=code), you'll note that high-contrast is an orthogonal feature, so they actually have defined normal and high-contrast variants for each of the colorblind themes. 

<img width="795" height="747" alt="Image" src="https://github.com/user-attachments/assets/b091450c-5f8b-4078-9762-a9e383e609a2" />

Let me know if there's anything I can do for you

**Comment by NinoScript** [comment]
Hey, for everyone looking for a solution, here's a gift for you:
https://gist.github.com/NinoScript/2dfb983de3bf5620beb8304ce653e05c

<img width="102" height="73" alt="Image" src="https://github.com/user-attachments/assets/9d5788c7-b528-412f-a50a-ab8c03c919f6" />
